src/core/agent/services/signifyNotificationService.ts

```diff
@@ -102,6 +102,24 @@ class SignifyNotificationService extends AgentService {
         notifications.notes.shift();
       }
       for (const notif of notifications.notes) {
+        if (notif.a.r === NotificationRoute.MultiSigIcp) {
+          const multisigNotification = await this.signifyClient
+            .groups()
+            .getRequest(notif.a.d);
+          const multisigId = multisigNotification[0]?.exn?.a?.gid;
+          if (!multisigId) {
+            continue;
+          }
+          const isMultisigInitiator =
+            await Agent.agent.multiSigs.isMultisigInitiator(multisigId);
+          const notificationForThisMultisig =
+            await Agent.agent.signifyNotifications.findNotificationByMultisigId(
+              multisigId
+            );
+          if (isMultisigInitiator && notificationForThisMultisig) {
```

Comment:
We just want 1 notification ever so I think it would be `isMultisigInitiator || notificationForThisMultisig`

Reply:
Hmmm, do we need to handle something when other parties confirm the joining of the multisig on the initiator side?

Reply:
Maybe more clear looking at https://github.com/cardano-foundation/cf-identity-wallet/pull/473#discussion_r1602798241

So I think it's more `if (multiSigExists || notificationForThisMultisig )`

So, if we receive a multi-sig inception message, we skip it if either:
1. The multisig exists in our wallet already (we initiated, or we already accepted notification and joined (notification deleted)), OR,
2. The notification is already in the queue for the user (e.g. if you get 2 icp notifications for this multi-sig as a receiver)